src/frontends/tensorflow/src/translate_session.cpp

```diff
@@ -0,0 +1,343 @@
+                    result_node->set_friendly_name(model_output_name);
+                    results.push_back(result_node);
+                }
+            } else if (port_type == "out") {
+                const auto& node_outputs = ng_op_map[operation_name];
+                FRONT_END_GENERAL_CHECK(node_outputs.size() > port_index,
+                                        "Output port with index " + std::to_string(port_index) + " of " +
+                                            operation_name + "node specified as custom output does not exist");
+                auto result_node = std::make_shared<ov::opset8::Result>(node_outputs[port_index]);
+                result_node->set_friendly_name(model_output_name);
+                results.push_back(result_node);
+            } else if (port_type == "in") {
+                // TODO: avoid this traversing by having a map for OpPlace objects, for example
+                std::shared_ptr<OpPlace> operation_place = nullptr;
+                for (const auto& op_place : operation_places) {
+                    FRONT_END_GENERAL_CHECK(!op_place->get_names().empty(), "No names for OpPlace found.");
+                    if (op_place->get_names()[0] == operation_name) {
+                        operation_place = op_place;
+                    }
+                }
+                FRONT_END_GENERAL_CHECK(operation_place, "There is no operation place with a name: " + operation_name);
+                auto operation_decoder = operation_place->get_decoder();
+
+                // get to know a producer node and by which its output port data is generated
+                std::string producer_name;
+                size_t producer_port_idx;
+                try {
+                    operation_decoder->get_input_node(port_index, producer_name, producer_port_idx);
+                } catch (const std::exception&) {
+                    FRONT_END_THROW("[ ERROR ] Exception happened when preparing input " + std::to_string(port_index) +
+                                    " for op '" + operation_decoder->get_op_name() + "', expected input name: '" +
+                                    producer_name +
+                                    "', expected input port index: " + std::to_string(producer_port_idx) + '\n');
+                }
+
+                // add Result node for this producer output port
+                const auto& node_outputs = ng_op_map[producer_name];
+                FRONT_END_GENERAL_CHECK(node_outputs.size() > producer_port_idx,
+                                        "Output port with index " + std::to_string(producer_port_idx) + " of " +
+                                            producer_name + "node specified as custom output does not exist");
+                auto result_node = std::make_shared<ov::opset8::Result>(node_outputs[producer_port_idx]);
+                result_node->set_friendly_name(model_output_name);
+                results.push_back(result_node);
+            }
```

Comment:
Suggest to add a message in case of unknown port type.

Reply:
This is a sort of the legacy code:) I didn't touch this in this PR and this is copy-pasted from frontend.cpp.
I would really give a chance to the code below to insert Results in this case.